src/rosdep2/platforms/pip.py

```diff
@@ -70,6 +84,41 @@ def get_pip_command():
     return None
 
 
+def externally_managed_installable():
+    """
+    PEP 668 enacted in Python 3.11 blocks pip from working in "externally
+    managed" envrionments such operating systems with included package
+    managers. If we're on Python 3.11 or greater, we need to check that pip
+    is configured to allow installing system-wide packages with the
+    flagrantly named "break system packages" config option or environment
+    variable.
+    """
+    if sys.version_info >= (3, 11):
+        if 'PIP_BREAK_SYSTEM_PACKAGES' in os.environ and os.environ[
+            'PIP_BREAK_SYSTEM_PACKAGES'
+        ].lower() in ('yes', '1', 'true'):
```

Comment:
minor suggestion, but, if you want, you could improve readability a little bit by changing it to something like:

```suggestion
        if (
            'PIP_BREAK_SYSTEM_PACKAGES' in os.environ
            and os.environ['PIP_BREAK_SYSTEM_PACKAGES'].lower() in ('yes', '1', 'true')
        ):
```

not actually sure if exact change passes linters, though.

Reply:
The only think flake8 got upset by was the and being on the continuing line rather than the end of the first. I like this form better thanks.

Reply:
This suggestion was incorporated into a2e9a1e

Reply:
I think you have an extra indentation level there:

```diff
     if (
-             'PIP_BREAK_SYSTEM_PACKAGES' in os.environ and
-             os.environ['PIP_BREAK_SYSTEM_PACKAGES'].lower() in ('yes', '1', 'true')
+         'PIP_BREAK_SYSTEM_PACKAGES' in os.environ and
+         os.environ['PIP_BREAK_SYSTEM_PACKAGES'].lower() in ('yes', '1', 'true')
     ):
         return True
```

in my experience the usual linters don't complain about this for some reason.

> The only think flake8 got upset by was the and being on the continuing line rather than the end of the first.

Ah yeah I always get bit by that!